File: sdk/python/entity/recon_order_main.py
from dataclasses import dataclass
from datetime import datetime
from typing import Optional

@dataclass
class ReconOrderMain:
    """对账订单主记录"""
    order_no: str
    pay_amount: float
    platform_income: float
    pay_fee: float
    split_total_amount: float
    
    # 状态枚举值 (0: PENDING, 1: SUCCESS, 2: FAILURE)
    pay_status: int
    split_status: int
    notify_status: int
    recon_status: int
    
    notify_result: Optional[str] = None
    
    create_time: Optional[datetime] = None
    update_time: Optional[datetime] = None
    
    # 退款相关
    refund_status: Optional[int] = None
    refund_amount: Optional[float] = None
    refund_time: Optional[datetime] = None

File: sdk/go/entity/recon_order_main.go
package entity

import (
	"time"

	"github.com/shopspring/decimal"
)

// ReconOrderMain 对账订单主记录
type ReconOrderMain struct {
	ID               int64           `json:"id" gorm:"primaryKey"`
	OrderNo          string          `json:"orderNo"`
	MerchantId       string          `json:"merchantId"`
	MerchantName     string          `json:"merchantName"`
	MerchantOrderNo  string          `json:"merchantOrderNo"`
	OrderAmount      decimal.Decimal `json:"orderAmount" gorm:"type:decimal(20,2)"`
	ActualAmount     decimal.Decimal `json:"actualAmount" gorm:"type:decimal(20,2)"`
	PlatformIncome   decimal.Decimal `json:"platformIncome" gorm:"type:decimal(20,2)"`
	PayFee           decimal.Decimal `json:"payFee" gorm:"type:decimal(20,2)"`
	SplitTotalAmount decimal.Decimal `json:"splitTotalAmount" gorm:"type:decimal(20,2)"`
	PayStatus        int             `json:"payStatus"`
	SplitStatus      int             `json:"splitStatus"`
	NotifyStatus     int             `json:"notifyStatus"`
	ReconStatus      int             `json:"reconStatus"`
	OrderTime        time.Time       `json:"orderTime"`
	PayTime          time.Time       `json:"payTime"`
	ReconTime        *time.Time      `json:"reconTime"`
	CreateTime       time.Time       `json:"createTime"`
	UpdateTime       time.Time       `json:"updateTime"`
}
